src/test/java/org/icgc/argo/program_service/services/EgoServiceIT.java

```diff
@@ -18,256 +18,240 @@
-      .collect(Collectors.toUnmodifiableList());
-    assertEquals(groupsLeft.size(),0);
+  public void getUser(){
+    stubFor(get(urlEqualTo(format("/users?query=%s", TEST_EMAIL)))
+            .willReturn(aResponse()
+                    .withStatus(OK.value())
+                    .withHeader("Content-Type", "application/json")
+                    .withBodyFile("getUser.json")));
+    assertTrue(client.getUser(TEST_EMAIL).isPresent());
+    assertEquals(client.getUser(TEST_EMAIL).get().getEmail(), TEST_EMAIL);
   }
 
   @Test
-  void egoServiceInitialization() {
-    assertNotNull(ReflectionTestUtils.getField(egoService, "egoPublicKey"));
+  public void joinProgram_success() {
+    val egoGroupId = UUID.randomUUID();
+    entityGenerator.setUpProgramEgoGroupEntity(SHORT_NAME, UserRole.COLLABORATOR, egoGroupId);
+
+    // Mock ego endpoint getUser
+    stubFor(get(urlEqualTo(format("/users?query=%s", TEST_EMAIL)))
+            .willReturn(aResponse()
+                    .withStatus(OK.value())
+                    .withHeader("Content-Type", "application/json")
+                    .withBodyFile("getUser.json")));
+    assertTrue(client.getUser(TEST_EMAIL).isPresent());
+    assertEquals(client.getUser(TEST_EMAIL).get().getEmail(), TEST_EMAIL);
+
+    //mock ego endpoint getUserByGroupId
+    stubFor(get(urlEqualTo(format("/groups/%s/users", egoGroupId)))
+            .willReturn(aResponse()
+                    .withStatus(OK.value())
+                    .withHeader("Content-Type", "application/json")
+                    .withBodyFile("userOne.json")));
+
+    // mock ego endpoint addUsersToGroup
+    stubFor(post(urlEqualTo(format("/groups/%s/users", egoGroupId)))
+            .willReturn(aResponse()
+                    .withStatus(OK.value())
+                    .withHeader("Content-Type", "application/json")
+                    .withBody("User is added to group.")));
+
+    // Call the target method
+    assertTrue(egoService.joinProgram(TEST_EMAIL, SHORT_NAME, UserRole.COLLABORATOR));
   }
 
   @Test
-  void getUser() {
-    val egoUser1 = client.getUser(TEST_EMAIL);
-    val egoUser2 = client.getUser(ADMIN_USER_EMAIL);
-    val egoUser3 = client.getUser(COLLABORATOR_USER_EMAIL);
-    assertTrue(egoUser1.isPresent());
-    assertTrue(egoUser2.isPresent());
-    assertTrue(egoUser3.isPresent());
+  public void join_program_user_not_found_fail(){
+    stubFor(get(urlEqualTo(format("/users?query=%s", TEST_EMAIL)))
+            .willReturn(aResponse()
+                    .withStatus(OK.value())
+                    .withHeader("Content-Type", "application/json")
+                    .withBodyFile("getUser_empty_result.json")));
+    assertTrue(client.getUser(TEST_EMAIL).isEmpty());
+    exceptionRule.expect(NotFoundException.class);
+    exceptionRule.expectMessage(TEST_EMAIL);
+    exceptionRule.expectMessage("user does not exist in ego");
+    egoService.joinProgram(TEST_EMAIL, SHORT_NAME, UserRole.COLLABORATOR);
```

Comment:
what is the point of this call?

Reply:
The call is to verify if `joinProgram` throws `NotFoundException` if user is not returned by ego.